### PR TITLE
Tag and TagType Model for the track and instruments.

### PR DESCRIPTION
**Pull request by prio101**
- [x] Added the database model and seeds for the instruments and tag_type.
- [x] User must add one instrument over the instrument table.
   - [x]  Added the API endpoint for the instruments over the component.
   - [x] User can add multiple tags / instruments.
   - [x] Have to introduce a JWT / Secuirity for the API actions for users. To avoid Spamming.
   
- [x] Track can have tags through TagType ( Genere, Request )

### CODE REVIEW

File: app/javascript/components/routes/Instrument.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+			.then(data => this.setState({ data: data.data }))
+			.catch(error => console.error(error));
+
+		// fetching the selected tags if any 
+		fetch(apiEndpoint+'/selected-instruments', {
+			method: 'POST',
+			headers: {
+				'Content-Type': 'application/json'
+			},
+			body: JSON.stringify({ user_id: 1 })
```
Comment (prio101) [inline app/javascript/components/routes/Instrument.js:78]: Here I am sending the HardCoded UserId: 1 for feature implementation.
We still have to introduce a JWT or some other way to get the authenticated user data. Need approval for task. @dezman

File: app/controllers/api/api_controller.rb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+class Api::ApiController < ApplicationController
+  skip_before_action :verify_authenticity_token
```
Comment (dezman) [inline app/controllers/api/api_controller.rb:2]: Do we have to do this?

File: app/controllers/api/v1/instruments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
+class Api::V1::InstrumentsController < Api::ApiController
+  def index
+    instrument_type = TagType.find_by(name: :instrument, published: true)
```
Comment (dezman) [inline app/controllers/api/v1/instruments_controller.rb:3]: You can create scopes on the `TagType` model for these.

File: app/javascript/components/Home.js [hunk 4]
```diff
@@ -98,6 +100,7 @@ class Home extends React.Component {
             <NavLink to="/">🎵 Fresh</NavLink>
             <NavLink to="/baked">🧁 Baked</NavLink>
             <NavLink to="/artists">🧞 Artists</NavLink>
+            <NavLink to="/instruments"> Instruments</NavLink>
```
Comment (dezman) [inline app/javascript/components/Home.js:103]: Lets not add this link here yet I think

File: app/javascript/components/routes/Instrument.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import _ from 'lodash';
+import { apiEndpoint, upcase, filterOutOldData, initialDataList } from '../util/helpers';
+import tw from "tailwind-styled-components"
+
+// styled components
+const Section = tw.section`
+	flex
```
Comment (dezman) [inline app/javascript/components/routes/Instrument.js:8]: This is awesome!!

File: app/javascript/components/routes/Instrument.js [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+	p-4
+	border
+	border-gray-300
+	rounded-lg
+	shadow-lg
+`
+
+
+class InstrumentTag extends React.Component {
+	constructor(props) {
```
Comment (dezman) [inline app/javascript/components/routes/Instrument.js:52]: Lets use 2 space indentation everywhere

File: app/javascript/components/routes/Instrument.js (hunk 1 above)
Comment (dezman) [reply app/javascript/components/routes/Instrument.js:78]: Sweet. Yea we can use ActiveStorage, I dont think we need a JWT.

Comment (dezman) [reply app/javascript/components/routes/Instrument.js:78]: Or rather can we talk about this, I don't think I understand.

**Review by dezman — CHANGES_REQUESTED** [review]
Looks awesome! Made a few comments. Lets work on the UI to create /  add these next and then add specs.

File: app/javascript/components/Home.js (hunk 4 above)
Comment (prio101) [reply app/javascript/components/Home.js:103]: okay

File: app/javascript/components/routes/Instrument.js (hunk 6 above)
Comment (prio101) [reply app/javascript/components/routes/Instrument.js:52]: Its already using 2 space indentation

File: app/javascript/components/routes/Instrument.js (hunk 1 above)
Comment (prio101) [reply app/javascript/components/routes/Instrument.js]: Solved

**Review by dezman — APPROVED** [review]

**Comment by dezman** [comment]
Awesome so fast!

**Pull request merged**